201-timeseriesinsights-environment-with-eventhub/azuredeploy.json

```diff
@@ -194,16 +208,50 @@
           },
           "tags": "[variables('eventSourceTagsValue')]",
           "dependsOn": [
-            "[concat('Microsoft.TimeSeriesInsights/environments/',parameters('environmentName'))]",
+            "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]",
             "[concat('Microsoft.EventHub/namespaces/', parameters('eventHubNamespaceName'), '/eventHubs/', parameters('eventHubName'), '/consumerGroups/', parameters('consumerGroupname'))]"
           ]
         }
       ]
+    },
+    {
+      "condition": "[if(equals(first(parameters('accessPolicyReaderObjectIds')), 'thisArrayIsReallyEmpty'), bool('false'), bool('true'))]",
+      "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+      "name": "[concat(parameters('environmentName'), '/', 'readerAccessPolicy', copyIndex())]",
+      "copy": { 
+        "name": "accessPolicyReaderCopy", 
+        "count": "[length(parameters('accessPolicyReaderObjectIds'))]"
```

Comment:
You're right we have a "bug" there...  for count let's use:

if(empty(parameters('array')), 1, length(parameters('array'))

The the rest of the items can be simplified per individual comments.

Reply:
Done